services/client/services.go

```diff
@@ -542,6 +579,10 @@ func validateAdvertisement(sid services.ServiceID, id string) error {
 	return nil
 }
 
+func leaderCacheKey(sid services.ServiceID, opts services.ElectionOptions) string {
+	return fmt.Sprintf("%s/%s", sid.String(), opts.String())
```

Comment:
Okay in that case I'd rename `String` to `ID` on the options just to be more explicit.